pkg/insight/insightstore/deploymentstore.go

```diff
@@ -17,26 +17,391 @@ package insightstore
 import (
 	"context"
 	"errors"
+	"fmt"
+	"time"
 
+	"google.golang.org/protobuf/proto"
+
+	"github.com/pipe-cd/pipecd/pkg/filestore"
+	"github.com/pipe-cd/pipecd/pkg/insight"
 	"github.com/pipe-cd/pipecd/pkg/model"
 )
 
 var (
-	errUnimplemented = errors.New("unimplemented")
+	errInvalidArg    = errors.New("invalid arg")
+	errLargeDuration = errors.New("too large duration")
+	errExceedMaxSize = errors.New("exceed max file size")
+	errInconsistent  = errors.New("no consistency between meta and chunk")
+)
+
+const (
+	maxChunkByteSize = 1 * 1024 * 1024
```

Comment:
nits, add a comment to show the size in human-readable format. Same with the maxDuration value